crates/rpc/rpc-builder/src/auth.rs

```diff
@@ -202,7 +202,11 @@ impl AuthServerConfig {
 
         let local_addr = server.local_addr()?;
         let handle = server.start(module.inner.clone());
-        let ipc_handle = ipc_server_config.build(ipc_endpoint.path())?.start(module.inner).await?;
+        let mut ipc_handle = None;
+        if let (Some(ipc_server_config), Some(ipc_endpoint)) = (ipc_server_config, ipc_endpoint) {
+            ipc_handle =
+                Some(ipc_server_config.build(ipc_endpoint.path())?.start(module.inner).await?);
```

Comment:
the setup for this is a bit odd for the regular rpc server as well

https://github.com/paradigmxyz/reth/blob/67558db6f39a9bc260cac6a88c614c6f61818437/crates/rpc/rpc-builder/src/lib.rs#L1738-L1745

I think what we actually want here is the same check, or change this to if let some endpoint